src/platforms/linux/linuxdependencies.cpp

```diff
@@ -156,3 +156,31 @@ QString LinuxDependencies::kdeFrameworkVersion() {
 
   return QString();
 }
+
+// static
+QString LinuxDependencies::desktopFileId(const QString& path) {
```

Comment:
This method seems to return a non-empty string even if '/applications/' is not in the path. Should it try to detect error conditions and return an empty string in those cases, or is that not possible? Callers seem to pass on the result to other methods without additional checks.

Reply:
It's not entirely clear how to handle that edge case, but I wanted to try and err on the side of caution and make sure that we always return something suitable for cgroup matching. I guess taking the file name and using that might be a more suitable fallback for that case.